apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -226,6 +226,20 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return undefined;
   }
 
+  public getShrinkwrapEntryFromTempProjectDependencyKey(
+    tempProjectDependencyKey: string
+  ): IPnpmShrinkwrapDependencyYaml | undefined {
+    return this._shrinkwrapJson.packages[tempProjectDependencyKey];
+  }
+
+  public getShrinkwrapEntry(name: string, version: string): IPnpmShrinkwrapDependencyYaml | undefined {
+    // Version can sometimes be in the form of a path that's already in the /name/version format.
+    const packageId: string = version.indexOf('/') !== -1
+      ? version
+      : `/${name}/${version}`;
```

Comment:
Not sure if this will work on pnpm 3.5+

Reply:
As discussed we will fix the pnpm 3.5 compatibility in my PR

Reply:
Let's address that in https://github.com/microsoft/web-build-tools/pull/1542